src/components/views/rooms/ThirdPartyMemberInfo.js

```diff
@@ -0,0 +1,142 @@
+            senderName: sender ? sender.name : this.props.event.getSender(),
+        };
+    }
+
+    componentWillMount(): void {
+        MatrixClientPeg.get().on("RoomState.events", this.onRoomStateEvents);
+    }
+
+    componentWillUnmount(): void {
+        const client = MatrixClientPeg.get();
+        if (client) {
+            client.removeListener("RoomState.events", this.onRoomStateEvents);
+        }
+    }
+
+    onRoomStateEvents = (ev) => {
+        if (ev.getType() === "m.room.third_party_invite" && ev.getStateKey() === this.state.stateKey) {
+            const newDisplayName = ev.getContent().display_name;
+            const isInvited = !!newDisplayName; // display_name indicates a valid invite
+
+            const newState = {invited: isInvited};
+            if (newDisplayName) newState['displayName'] = newDisplayName;
+            this.setState(newState);
+        }
+    };
+
+    onCancel = () => {
+        dis.dispatch({
+            action: "view_3pid_invite",
+            event: null,
+        });
+    };
+
+    onKickClick = () => {
+        MatrixClientPeg.get().sendStateEvent(this.state.roomId, "m.room.third_party_invite", {}, this.state.stateKey)
+            .catch((err) => {
+                console.error(err);
+
+                // Revert echo because of error
+                this.setState({invited: true});
+
+                const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
+                Modal.createTrackedDialog('Revoke 3pid invite failed', '', ErrorDialog, {
+                    title: _t("Failed to revoke invite"),
+                    description: _t(
+                        "Could not revoke the invite. The server may be experiencing a temporary problem or " +
+                        "you do not have sufficient permissions to revoke the invite.",
+                    ),
+                });
+            });
+
+        // Local echo
+        this.setState({invited: false});
+    };
+
+    render() {
+        const AccessibleButton = sdk.getComponent("elements.AccessibleButton");
+
+        let adminTools = null;
+        if (this.state.canKick && this.state.invited) {
+            adminTools = (
+                <div className="mx_MemberInfo_container">
+                    <h3>{_t("Admin Tools")}</h3>
+                    <div className="mx_MemberInfo_buttons">
+                        <AccessibleButton className="mx_MemberInfo_field" onClick={this.onKickClick}>
+                            {_t("Revoke invite")}
+                        </AccessibleButton>
+                    </div>
+                </div>
+            );
+        }
+
+        // We shamelessly rip off the MemberInfo styles here.
+        return (
+            <div className="mx_MemberInfo">
+                <div className="mx_MemberInfo_name">
+                    <AccessibleButton className="mx_MemberInfo_cancel"
+                                      onClick={this.onCancel}
```

Comment:
Nit: I think we concluded we shouldn't align attributes like this. Instead the ones on new lines are 4 space indented.

Reply:
Yup, good catch. I really need to find the option in my ide to not do this